diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
index 338ff85..0b738ea 100644
--- a/cmake/Config.cmake.in
+++ b/cmake/Config.cmake.in
@@ -28,7 +28,7 @@ if(@GTSAM_USE_TBB@)
 endif()
 
 if(@GTSAM_USE_SYSTEM_EIGEN@)
-find_dependency(Eigen3 REQUIRED)
+find_dependency(Eigen3 REQUIRED CONFIG)
 endif()
 
 # Load exports
diff --git a/cmake/GtsamBuildTypes.cmake b/cmake/GtsamBuildTypes.cmake
index cf0dfdf..75c390e 100644
--- a/cmake/GtsamBuildTypes.cmake
+++ b/cmake/GtsamBuildTypes.cmake
@@ -131,7 +131,7 @@ else()
   endif()
 
   set(GTSAM_COMPILE_OPTIONS_PRIVATE_COMMON
-    -Werror                                        # Enable warnings as errors
+    #-Werror                                        # Enable warnings as errors
     -Wall                                          # Enable common warnings
     -Wpedantic                                     # Enable pedantic warnings
     $<$<COMPILE_LANGUAGE:CXX>:-Wextra -Wno-unused-parameter> # Enable extra warnings, but ignore no-unused-parameter (as we ifdef out chunks of code)
diff --git a/cmake/HandleBoost.cmake b/cmake/HandleBoost.cmake
index 0325112..f6a73f5 100644
--- a/cmake/HandleBoost.cmake
+++ b/cmake/HandleBoost.cmake
@@ -23,12 +23,12 @@ endif()
 
 # Store these in variables so they are automatically replicated in GTSAMConfig.cmake and such.
 set(BOOST_FIND_MINIMUM_VERSION 1.65)
-set(BOOST_FIND_MINIMUM_COMPONENTS serialization system filesystem thread program_options date_time timer chrono regex)
+set(BOOST_FIND_MINIMUM_COMPONENTS serialization filesystem thread program_options date_time timer chrono regex)
 
 find_package(Boost ${BOOST_FIND_MINIMUM_VERSION} COMPONENTS ${BOOST_FIND_MINIMUM_COMPONENTS} REQUIRED)
 
 # Required components
-if(NOT Boost_SERIALIZATION_LIBRARY OR NOT Boost_SYSTEM_LIBRARY OR NOT Boost_FILESYSTEM_LIBRARY OR
+if(NOT Boost_SERIALIZATION_LIBRARY OR NOT Boost_FILESYSTEM_LIBRARY OR
     NOT Boost_THREAD_LIBRARY OR NOT Boost_DATE_TIME_LIBRARY)
   message(FATAL_ERROR "Missing required Boost components >= v1.65, please install/upgrade Boost or configure your search paths.")
 endif()
@@ -37,7 +37,6 @@ option(GTSAM_DISABLE_NEW_TIMERS "Disables using Boost.chrono for timing" OFF)
 # Allow for not using the timer libraries on boost < 1.48 (GTSAM timing code falls back to old timer library)
 set(GTSAM_BOOST_LIBRARIES
   Boost::serialization
-  Boost::system
   Boost::filesystem
   Boost::thread
   Boost::date_time
