diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
index e6d165ed5..b36c5a95e 100644
--- a/cmake/Config.cmake.in
+++ b/cmake/Config.cmake.in
@@ -32,7 +32,7 @@ if(@GTSAM_USE_TBB@)
 endif()
 
 if(@GTSAM_USE_SYSTEM_EIGEN@)
-find_dependency(Eigen3 REQUIRED)
+find_dependency(Eigen3 CONFIG REQUIRED)
 endif()
 
 # Load exports
diff --git a/cmake/HandleMKL.cmake b/cmake/HandleMKL.cmake
index 07ac53e4f..fde0185b6 100644
--- a/cmake/HandleMKL.cmake
+++ b/cmake/HandleMKL.cmake
@@ -38,5 +38,8 @@ if(WIN32 AND GTSAM_USE_EIGEN_MKL AND DEFINED VCPKG_INSTALLED_DIR)
         COMMAND_EXPAND_LISTS
         VERBATIM
     )
+	if (NOT TARGET check)
+		add_custom_target(check)
+	endif()
     add_dependencies(check copy_mkl_dlls)
 endif()
