diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9660f55..236a704 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,7 @@ set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib)
 
 OPTION(BUILD_TESTS "Build tests" ON)
 OPTION(BUILD_PYTHON "Build Python extension" OFF)
+OPTION(BUILD_WITH_MARCHNATIVE "Build with -march=native" ON)
 
 IF(MSVC)
   set(BUILD_SHARED_LIBS OFF)
@@ -35,7 +36,7 @@ ELSE()
   ELSEIF (CMAKE_SYSTEM_PROCESSOR MATCHES 
           "(arm)|(ARM)|(armhf)|(ARMHF)|(armel)|(ARMEL)")
     add_definitions (-march=armv7-a)
-  ELSE ()
+  ELSEIF (BUILD_WITH_MARCHNATIVE)
     add_definitions (-march=native) #TODO use correct c++11 def once everybody has moved to gcc 4.7 # for now I even removed std=gnu++0x
   ENDIF()
   add_definitions (
@@ -54,8 +55,7 @@ IF(BUILD_POSITION_INDEPENDENT_CODE)
 ENDIF()
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/modules/")
-find_package(Eigen REQUIRED)
-set(ADDITIONAL_INCLUDE_DIRS ${EIGEN_INCLUDE_DIRS} ${EIGEN_INCLUDE_DIR}/unsupported)
+find_package(Eigen3 REQUIRED)
 
 set( OPENGV_SOURCE_FILES
   src/absolute_pose/modules/main.cpp
@@ -189,11 +189,8 @@ target_include_directories( opengv PUBLIC
                 # only when building from the source tree
                 $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
                 # only when using the lib from the install path
-                $<INSTALL_INTERFACE:include>
-                ${ADDITIONAL_INCLUDE_DIRS} )
-
-target_include_directories( random_generators PRIVATE ${ADDITIONAL_INCLUDE_DIRS} )
-
+                $<INSTALL_INTERFACE:include> )
+target_link_libraries(opengv PUBLIC Eigen3::Eigen)
 target_link_libraries( random_generators opengv )
 
 IF (BUILD_TESTS)
diff --git a/modules/Config.cmake.in b/modules/Config.cmake.in
index 9b4c9ee..485720c 100644
--- a/modules/Config.cmake.in
+++ b/modules/Config.cmake.in
@@ -1,4 +1,4 @@
 @PACKAGE_INIT@
-
+find_dependency(Eigen3)
 include("${CMAKE_CURRENT_LIST_DIR}/@targets_export_name@.cmake")
 check_required_components("@PROJECT_NAME@")
diff --git a/src/absolute_pose/modules/main.cpp b/src/absolute_pose/modules/main.cpp
index ed0c271..a179aaf 100644
--- a/src/absolute_pose/modules/main.cpp
+++ b/src/absolute_pose/modules/main.cpp
@@ -29,8 +29,8 @@
  ******************************************************************************/
 
 
-#include <Eigen/NonLinearOptimization>
-#include <Eigen/NumericalDiff>
+#include <unsupported/Eigen/NonLinearOptimization>
+#include <unsupported/Eigen/NumericalDiff>
 
 #include <opengv/absolute_pose/modules/main.hpp>
 #include <opengv/absolute_pose/modules/gp3p/modules.hpp>
 