diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9dabfd0..7b73418 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,9 +128,9 @@ endif()
 #--------------------
 # DEPENDENCY:  boost
 #--------------------
-find_package(Boost REQUIRED COMPONENTS thread system program_options date_time)
+find_package(Boost REQUIRED COMPONENTS thread program_options date_time)
 if (Boost_MINOR_VERSION GREATER 47)
-	find_package(Boost REQUIRED COMPONENTS thread system program_options date_time chrono)
+	find_package(Boost REQUIRED COMPONENTS thread program_options date_time chrono)
 endif ()

 #--------------------
diff --git a/libpointmatcherConfig.cmake.in b/libpointmatcherConfig.cmake.in
index 6de2c85..7c887fd 100644
--- a/libpointmatcherConfig.cmake.in
+++ b/libpointmatcherConfig.cmake.in
@@ -7,9 +7,9 @@
 include(CMakeFindDependencyMacro)
 find_dependency(libnabo REQUIRED)
 find_dependency(yaml-cpp REQUIRED)
-find_package(Boost COMPONENTS thread system program_options date_time REQUIRED)
+find_package(Boost COMPONENTS thread program_options date_time REQUIRED)
 if (Boost_MINOR_VERSION GREATER 47)
-  find_package(Boost COMPONENTS thread system program_options date_time chrono REQUIRED)
+  find_package(Boost COMPONENTS thread program_options date_time chrono REQUIRED)
 endif ()
 include(${CMAKE_CURRENT_LIST_DIR}/libpointmatcher-config.cmake)

